### PR TITLE
Added SlideUp Search Bar, Navigation, Re-organised folders

### PR DESCRIPTION
**Pull request by cam-arch**

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,33 +1,78 @@
 import { ApolloClient, ApolloProvider, InMemoryCache } from '@apollo/client'
-import BusinessView from '@components/BusinessView'
+
+// Import Navigation Libraries
```
Comment (cg-2611) [inline src/App.tsx]: no need to comment imports

File: src/components/AccountButton.tsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        <React.Fragment>
+            <View style={[styles.circleInside, {marginLeft: screenWidth-80}]}>
+                <View style={styles.circleHead}/>
+                <View style={styles.ovalBody}/>
+            </View>
+        </React.Fragment>
+        
+    )
+
+
```
Comment (cg-2611) [inline src/components/AccountButton.tsx]: remove unnecessary whitespace

File: src/components/AccountView.tsx [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+import {View, StyleSheet, ScrollView} from 'react-native';
+
+const AccountView = () => {
+
+    return (
+        <ScrollView>
+        </ScrollView>
+        
+    )
+
```
Comment (cg-2611) [inline src/components/AccountView.tsx]: remove whitespace

File: src/components/Login/SignUpViews/LoginView.tsx [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+const LoginView = () => {
+
+    return (
+        <View>
+        </View>
+        
+    )
+
+
```
Comment (cg-2611) [inline src/components/Login/SignUpViews/LoginView.tsx]: remove whitespace

File: src/components/Login/SignUpViews/SignUpView.tsx [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+    return (
+        <View>
+        </View>
+        
+    )
+
+
+
+}
```
Comment (cg-2611) [inline src/components/Login/SignUpViews/SignUpView.tsx:6]: remove whitespace

File: src/components/MapSlideUpSheet.js [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+            <View style={styles.container}>
+              <SearchBar/>
+              <SearchView/>
+            </View>
+          </View>
+        </SlidingUpPanel>
+      </React.Fragment>
+    )
+  }
+
```
Comment (cg-2611) [inline src/components/MapSlideUpSheet.js]: remove whitespace

File: src/components/MapSlideUpSheet.js [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+    backgroundColor: '#2b8a3e',
+    width: 48,
+    height: 48,
+    padding: 8,
+    borderRadius: 24,
+    zIndex: 1
+  }
+})
+
+export default MapSlideUpSheet
```
Comment (cg-2611) [inline src/components/MapSlideUpSheet.js]: add new line at end of file

File: src/components/SearchView.tsx [hunk 8]
```diff
@@ -15,25 +15,28 @@ const TEST_BUSINESSES_DATA: DisplayableBusiness[] = [
...
       onPress={ () => {
         // Respond to business seletion tap here
+        navigation.push("BusinessView")
       }}
     />
   )
   return (
-    <SafeAreaView style={styles.searchList}>
+    <React.Fragment>
+    <ScrollView style={styles.searchList}>
```
Comment (cg-2611) [inline src/components/SearchView.tsx]: indentation

**Review by cg-2611 — CHANGES_REQUESTED** [review]

File: src/App.tsx (hunk 1 above)
Comment (cam-arch) [reply src/App.tsx]: k

File: src/components/AccountButton.tsx (hunk 2 above)
Comment (cam-arch) [reply src/components/AccountButton.tsx]: k

File: src/components/AccountView.tsx (hunk 3 above)
Comment (cam-arch) [reply src/components/AccountView.tsx]: k

File: src/components/Login/SignUpViews/LoginView.tsx (hunk 4 above)
Comment (cam-arch) [reply src/components/Login/SignUpViews/LoginView.tsx]: k

File: src/components/Login/SignUpViews/SignUpView.tsx (hunk 5 above)
Comment (cam-arch) [reply src/components/Login/SignUpViews/SignUpView.tsx:6]: k

File: src/components/MapSlideUpSheet.js (hunk 6 above)
Comment (cam-arch) [reply src/components/MapSlideUpSheet.js]: k

File: src/components/MapSlideUpSheet.js (hunk 7 above)
Comment (cam-arch) [reply src/components/MapSlideUpSheet.js]: k

File: src/components/SearchView.tsx (hunk 8 above)
Comment (cam-arch) [reply src/components/SearchView.tsx]: k

File: package.json [hunk 9]
```diff
@@ -13,6 +13,9 @@
   },
   "dependencies": {
     "@apollo/client": "^3.5.8",
+    "@react-navigation/native": "^6.0.7",
+    "@react-navigation/native-stack": "^6.3.0",
```
Comment (bridges-wood) [inline package.json:17]: This dependency is not used as far as I can tell?

File: src/App.tsx [hunk 10]
```diff
@@ -1,9 +1,20 @@
...
+import AccountView from "@components/AccountView";
+
+import AccountButton from "./components/AccountButton";
+import React from "react";
+import { StyleSheet, View, SafeAreaView, TouchableOpacity } from "react-native";
 import { Business, Location } from '@futureproof/typings'
 import Constants from 'expo-constants'
-import React from 'react'
-import { StyleSheet, View } from 'react-native'
+
```
Comment (bridges-wood) [inline src/App.tsx]: Remove blank lines in the middle of imports

File: package.json [hunk 11]
```diff
@@ -23,10 +26,16 @@
     "react-dom": "17.0.1",
     "react-native": "0.64.3",
     "react-native-elements": "^3.4.2",
+    "react-native-gesture-handler": "^2.2.0",
```
Comment (bridges-wood) [inline package.json:29]: Is this dependency used?

File: package.json [hunk 12]
```diff
@@ -23,10 +26,16 @@
...
     "react-native-image-viewing": "^0.2.1",
     "react-native-maps": "^0.29.4",
+    "react-native-navigation": "^7.24.3-hotfix.1",
```
Comment (bridges-wood) [inline package.json:32]: Is this dependency used?

File: package.json [hunk 13]
```diff
@@ -23,10 +26,16 @@
...
+    "react-native-reanimated": "^2.4.1",
```
Comment (bridges-wood) [inline package.json:33]: Is this dependency used?

File: package.json [hunk 14]
```diff
@@ -23,10 +26,16 @@
...
     "react-native-safe-area-context": "^3.4.1",
-    "react-navigation": "^4.4.4"
+    "react-native-screens": "^3.12.0",
+    "react-native-web": "0.17.1",
```
Comment (bridges-wood) [inline package.json]: Are these dependencies used?

File: src/components/AccountView.tsx [hunk 15]
```diff
@@ -0,0 +1,10 @@
+import React from "react";
+import { View, StyleSheet, ScrollView } from "react-native";
+
+const AccountView = () => {
+  return <ScrollView></ScrollView>;
+};
```
Comment (bridges-wood) [inline src/components/AccountView.tsx]: Why is there an empty component here?

File: src/components/BusinessViews/BusinessView.tsx [hunk 16]
```diff
@@ -0,0 +1,46 @@
+import DescriptionView from "@components/BusinessViews/DescriptionView";
+import ImagesCarousel from "@components/BusinessViews/ImagesCarousel";
+import RatingsView from "@components/BusinessViews/RatingsView";
+import TitleView from "@components/BusinessViews/TitleView";
+import { BusinessType, DisplayableBusiness } from "@futureproof/typings";
+import React from "react";
+import { SafeAreaView, ScrollView, StyleSheet } from "react-native";
+
+// PLACEHOLDER BUSINESS -> Update to fetch graphQL
```
Comment (bridges-wood) [inline src/components/BusinessViews/BusinessView.tsx:9]: Now that we have a solution to this problem, it would be a great time to get this working.

File: src/components/BusinessViews/ImagesCarousel.tsx [hunk 17]
```diff
@@ -0,0 +1,180 @@
...
+  },
+  {
+    uri: Image3,
+  },
+  {
+    uri: Image4,
+  },
+];
+
+const ImagesCarousel = () => {
```
Comment (bridges-wood) [inline src/components/BusinessViews/ImagesCarousel.tsx]: It would be good if this component took an array of images as one of its props.

File: src/components/Login/SignUpViews/LoginView.tsx [hunk 18]
```diff
@@ -0,0 +1,10 @@
+import React from "react";
+import { View, StyleSheet } from "react-native";
+
+const LoginView = () => {
+  return <View></View>;
+};
+
+const styles = StyleSheet.create({});
+
+export default LoginView;
```
Comment (bridges-wood) [inline src/components/Login/SignUpViews/LoginView.tsx]: Why is there an empty component here?

File: src/components/Login/SignUpViews/SignUpView.tsx [hunk 19]
```diff
@@ -0,0 +1,10 @@
+import React from "react";
+import { View, StyleSheet } from "react-native";
+
+const SignUpView = () => {
+  return <View></View>;
+};
+
+const styles = StyleSheet.create({});
+
+export default SignUpView;
```
Comment (bridges-wood) [inline src/components/Login/SignUpViews/SignUpView.tsx]: Why is there an empty component here?

File: src/components/MapSlideUpSheet.js [hunk 20]
```diff
@@ -0,0 +1,80 @@
...
+    backgroundColor: "#2b8a3e",
+    width: 48,
+    height: 48,
+    padding: 8,
+    borderRadius: 24,
+    zIndex: 1,
+  },
+});
+
+export default MapSlideUpSheet;
```
Comment (bridges-wood) [inline src/components/MapSlideUpSheet.js]: Why is this a `js` file? This project is written in TypeScript

File: src/components/MapSlideUpSheet.js [hunk 21]
```diff
@@ -0,0 +1,80 @@
...
+  TouchableOpacity,
+} from "react-native";
+import SlidingUpPanel from "rn-sliding-up-panel";
+import SearchBar from "@components/SearchBarView";
+import SearchView from "@components/SearchView";
+import SearchResultSorter from "@components/SortSearchResults/SearchResultSorter";
+
+const { height } = Dimensions.get("window");
+
+class MapSlideUpSheet extends React.Component {
```
Comment (bridges-wood) [inline src/components/MapSlideUpSheet.js:10]: We don't use class based components, only functional ones.

File: src/components/SearchBarView.tsx [hunk 22]
```diff
@@ -1,84 +1,91 @@
...
+      id
+      name
+      profilePicture
+      sustainabilityScore
+      customerScore
     }
-}
-`
+  }
+`;
```
Comment (bridges-wood) [inline src/components/SearchBarView.tsx]: This defines a fragment, not a query. I don't believe this will return a result.

File: src/components/SearchBarView.tsx [hunk 23]
```diff
@@ -1,84 +1,91 @@
...
 
 const SearchBarView = () => {
-  const [searchText, onChangeText] = useState('')
-    
-  const [executeSearch, { error }] = useLazyQuery(    // 'loading' and 'data' can also be returned (not just error)
+  const [searchText, onChangeText] = useState("");
+
+  const [executeSearch, { error }] = useLazyQuery(
+    // 'loading' and 'data' can also be returned (not just error)
```
Comment (bridges-wood) [inline src/components/SearchBarView.tsx:26]: This is a redundant comment.

File: src/components/SearchBarView.tsx [hunk 24]
```diff
@@ -1,84 +1,91 @@
...
+        (data) => console.log(data),
+        () => console.error(error)
+      );
+    }, 300);
+    return () => clearTimeout(delayDebounceFn);
+  }, [searchText]);
 
   return (
-    <SafeAreaView style={styles.safeAreaView}>
+    <React.Fragment>
```
Comment (bridges-wood) [inline src/components/SearchBarView.tsx:44]: Why has this changed from a SafeAreaView?

File: src/components/SearchView.tsx [hunk 25]
```diff
@@ -1,47 +1,51 @@
...
-  }
-]
+    profilePicture:
+      "https://upload.wikimedia.org/wikipedia/en/thumb/d/d3/Starbucks_Corporation_Logo_2011.svg/1200px-Starbucks_Corporation_Logo_2011.svg.png",
+  },
+];
 
-const SearchView = () => {
-  const [selectedId, setSelectedId] = useState(null)
+const SearchView = ({ navigation }: { navigation: any }) => {
```
Comment (bridges-wood) [inline src/components/SearchView.tsx]: Navigation should not have an `any` type.

File: src/components/SearchView.tsx [hunk 26]
```diff
@@ -1,47 +1,51 @@
...
+  const [selectedId, setSelectedId] = useState(null);
 
-  const renderSearchItem = ({ item } : { item : DisplayableBusiness}) => (
+  const renderSearchItem = ({ item }: { item: DisplayableBusiness }) => (
     <SearchResult
       business={item}
-      onPress={ () => {
+      onPress={() => {
         // Respond to business seletion tap here
```
Comment (bridges-wood) [inline src/components/SearchView.tsx:26]: This comment should be removed.

File: src/components/SearchView.tsx (hunk 25 above)
Comment (bridges-wood) [inline src/components/SearchView.tsx]: Can this component make a GraphQL query to get results

File: src/components/SortSearchResults/SearchResultSorter.tsx [hunk 27]
```diff
@@ -0,0 +1,112 @@
...
+  SafeAreaView,
+  StyleSheet,
+  Text,
+  TextStyle,
+  View,
+  ViewStyle,
+} from "react-native";
+import { sortOption as sortBy } from "../../cache";
+
+const INITIAL_OPTION_INDEX = 4; // config variable
```
Comment (bridges-wood) [inline src/components/SortSearchResults/SearchResultSorter.tsx]: Remove redundant comment.

File: src/components/SortSearchResults/SortOptions.tsx [hunk 28]
```diff
@@ -0,0 +1,114 @@
...
+ */
+const SortOptions = ({ options, initial, onChange }: ComponentProps) => {
+  let initialOption = null;
+
+  if (initial !== undefined && initial !== null) {
+    initialOption =
+      initial >= 0 && initial < options.length ? options[initial] : null;
+  }
+
+  // const io = !initial ? ((initial >= 0 && initial < options.length) ? options[initial] : null) : null
```
Comment (bridges-wood) [inline src/components/SortSearchResults/SortOptions.tsx:32]: Remove unnecessary comment

**Review by bridges-wood — CHANGES_REQUESTED** [review]
Quite a few comments here. Some that didn't really fit attached to any particular file:
- You need to run the linter on this code. Most of the changes here are style changes that are inconsistent with our established conventions.
- Why are some components now in new directories with names that do not follow the established naming convention?

File: package.json (hunk 12 above)
Comment (cam-arch) [reply package.json:32]: Yes, for the stack navigator

File: package.json (hunk 13 above)
Comment (cam-arch) [reply package.json:33]: Yes, for the stack navigator

File: package.json (hunk 14 above)
Comment (cam-arch) [reply package.json]: Yes, for the stack navigator

File: src/components/AccountView.tsx (hunk 15 above)
Comment (cam-arch) [reply src/components/AccountView.tsx]: I needed to setup the account page to enable the navigation between the pages. As it is only sprint 2, I haven't filled it in yet.

File: src/components/BusinessViews/ImagesCarousel.tsx (hunk 17 above)
Comment (cam-arch) [reply src/components/BusinessViews/ImagesCarousel.tsx]: I agree. I was waiting for the everyone's different work to be amalgamated first though.

File: src/components/Login/SignUpViews/LoginView.tsx (hunk 18 above)
Comment (cam-arch) [reply src/components/Login/SignUpViews/LoginView.tsx]: Regarding empty views, this is intentional as I wanted to get some of our additional pages laid out ready for me to add to the navigation. As we’re not on sprint 3 yet I have started work on the pages themselves yet though.

File: src/components/Login/SignUpViews/SignUpView.tsx (hunk 19 above)
Comment (cam-arch) [reply src/components/Login/SignUpViews/SignUpView.tsx]: Regarding empty views, this is intentional as I wanted to get some of our additional pages laid out ready for me to add to the navigation. As we’re not on sprint 3 yet I have started work on the pages themselves yet though.

File: src/components/MapSlideUpSheet.js (hunk 20 above)
Comment (cam-arch) [reply src/components/MapSlideUpSheet.js]: I was having issues with _draggedValue and _panel in TypeScript due to all the extra checks it does. This was a temporary fix to resolve the error and I will come back to it later.

File: src/components/MapSlideUpSheet.js (hunk 21 above)
Comment (cam-arch) [reply src/components/MapSlideUpSheet.js:10]: What's the problem with a class based component?

File: src/components/SearchBarView.tsx (hunk 22 above)
Comment (cam-arch) [reply src/components/SearchBarView.tsx]: Not my section.

File: src/components/SearchBarView.tsx (hunk 23 above)
Comment (cam-arch) [reply src/components/SearchBarView.tsx:26]: Not my section.

File: src/components/SearchBarView.tsx (hunk 24 above)
Comment (cam-arch) [reply src/components/SearchBarView.tsx:44]: So that it can be embedded properly in the slide up panel.

File: src/components/SearchView.tsx (hunk 26 above)
Comment (cam-arch) [reply src/components/SearchView.tsx:26]: Not my section.

File: src/components/SearchView.tsx (hunk 25 above)
Comment (cam-arch) [reply src/components/SearchView.tsx]: Not my section.

File: src/components/SortSearchResults/SearchResultSorter.tsx (hunk 27 above)
Comment (cam-arch) [reply src/components/SortSearchResults/SearchResultSorter.tsx]: Not my section.

File: src/components/SortSearchResults/SortOptions.tsx (hunk 28 above)
Comment (cam-arch) [reply src/components/SortSearchResults/SortOptions.tsx:32]: Not my section.

File: package.json (hunk 11 above)
Comment (cam-arch) [reply package.json:29]: Yes, for the slide up panel.

**Review by bridges-wood — CHANGES_REQUESTED** [review]
- Run npm lint
- Write tests for your new components

File: package.json (hunk 9 above)
Comment (Matthew5002) [reply package.json:17]: Have you resolved this?

File: src/App.spec.tsx [hunk 29]
```diff
@@ -0,0 +1,17 @@
+import { FeedScreen } from '@src/App'
+import { render, fireEvent } from '@testing-library/react-native'
+import React from 'react'
+
+it('renders default elements', () => {
+  //const { } = render(<FeedScreen />)
```
Comment (Matthew5002) [inline src/App.spec.tsx]: Did you add this? (I've not got this from my last pull of sprint 2 branch) Can this be removed if it's not doing anything?

File: src/components/BusinessViews/ImagesCarousel.tsx (hunk 17 above)
Comment (Matthew5002) [reply src/components/BusinessViews/ImagesCarousel.tsx]: I don't think my businessview changes will cause you any issue with making this change, as long as you just change the props for this view and give it a static input/parameters where it's used for now?

File: src/components/MapSlideUpSheet.js (hunk 21 above)
Comment (Matthew5002) [reply src/components/MapSlideUpSheet.js:10]: ^^ this does make sense for consistency and said we would stick to functions at the start but idm if it's a pain to change.

File: src/components/SearchBarView.tsx (hunk 24 above)
Comment (Matthew5002) [reply src/components/SearchBarView.tsx:44]: Yeah this doesn't need a safe area view anyway?

File: src/components/SearchView.tsx (hunk 25 above)
Comment (Matthew5002) [reply src/components/SearchView.tsx]: Can you quickly research this because it should have a type? I think I found it before when I was testing implementing navigation during sprint 1.

**Review by Matthew5002 — CHANGES_REQUESTED** [review]
Looks great, just some comments. I've resolved most of the comments for you but there are a few where they aren't resolved or I agree with Max, but really minor issues.

File: package.json (hunk 9 above)
Comment (cam-arch) [reply package.json:17]: Needed for the stack navigator

**Review by cg-2611 — APPROVED** [review]
just make sure that there are no linting errors before merge
